src/caret_analyze/infra/lttng/lttng.py

```diff
@@ -358,83 +400,106 @@ def _parse_lttng_data(
-            for event in event_collection:
-                event_name = event[LttngEventFilter.NAME]
-                if event_name in init_event_names:
-                    init_events.append(event)
-                else:
-                    run_events.append(event)
+            trace_dir_or_events = [trace_dir_or_events]
 
-            import functools
-            init_events.sort(key=functools.cmp_to_key(Lttng._compare_init_event))
-            handler.create_init_handler_map()
-            for event in tqdm(
-                    iter(init_events),
-                    total=len(init_events),
-                    desc='loading',
-                    mininterval=1.0):
-                event_name = event[LttngEventFilter.NAME]
-                handler_ = handler.handler_map[event_name]
-                handler_(event)
+        if len(trace_dir_or_events) == 0:
+            raise InvalidArgumentError('trace_dir_or_events should not be an empty list.')
 
-            handler.create_runtime_handler_map()
-            filtered_event_count = 0
-            for event in tqdm(
-                    iter(run_events),
-                    total=len(run_events),
-                    desc='loading',
-                    mininterval=1.0):
-                if len(event_filters) > 0 and \
-                        any(not f.accept(event, common) for f in event_filters):
-                    # memo:
-                    # case1 : arch = Architecture('lttng', tracing_log_path)
-                    # since event_filters is set,
-                    # initialization-related trace events are processed,
-                    # but all runtime-related trace events are not processed.
-                    # case2 : lttng = Lttng(tracing_log_path)
-                    # event_filters are not set, so all events are processed
-                    continue
-                if store_events:
-                    event_dict = {
-                        k: get_field(event, k) for k in event
-                    }
-                    events.append(event_dict)
-                filtered_event_count += 1
-                event_name = event[LttngEventFilter.NAME]
-                handler_ = handler.handler_map[event_name]
-                handler_(event)
+        # validate list[str] case
+        first_element_type = type(trace_dir_or_events[0])
+        if len(trace_dir_or_events) != len(Util.filter_items(
+                lambda x: isinstance(x, first_element_type), trace_dir_or_events)):
+            raise InvalidArgumentError(f'all elements of {trace_dir_or_events} \
+                                         must be same type: {first_element_type}.')
 
+        # TODO(hsgwa): Same implementation duplicated. Refactoring required.
+        if isinstance(trace_dir_or_events[0], str):
+            tid_remapper = MultiHostIdRemapper(LttngEventFilter.VTID)
+            pid_remapper = MultiHostIdRemapper(LttngEventFilter.VPID)
+            evant_remapper = AddressRemapper()
```

Comment:
Minor fix of typo.
```suggestion
            event_remapper = AddressRemapper()
```

Reply:
Fixed in [c67c785](https://github.com/tier4/caret_analyze/pull/455/commits/c67c7853ecf19972daffb03d6c53bbd6444449a8).